app/client/lib/ethereum/observeEvents.js

```diff
@@ -12,15 +14,31 @@ observeEvents = function(){
     */
     collectionObservers[collectionObservers.length] = Events.find({}).observe({
         /**
-        This will observe the transactions creation and create watchers for outgoing transactions, to see when they are mined.
+        This will observe when events are added and link it to the custom contract.
 
         @method added
         */
         added: function(newDocument) {
+            
+            // This creates a temporary cache for the contracts, to reduce the amount of db reads            
```

Comment:
From what I can see it looks like `update()` is only called if the `_id` is not already stored inside `contractEvents`. And this check is performed by loading the custom contract once using `findOne()` and cacheing it. So I agree, it's limiting the no. of db writes.